app/javascript/packages/document-capture/hooks/use-log-camera-info.ts

```diff
@@ -0,0 +1,69 @@
+import { useState, useEffect, useContext } from 'react';
+import AnalyticsContext from '../context/analytics';
+
+function getConstraints(deviceId) {
+  return {
+    video: {
+      width: {
+        ideal: 999999,
+      },
+      height: {
+        ideal: 999999,
+      },
+      deviceId: {
+        exact: deviceId,
+      },
+    },
+  };
+}
+
+function getCameraInfo(videoTrack) {
+  const cameraInfo = {
+    label: videoTrack.label,
+    frameRate: videoTrack.getSettings().frameRate,
+    height: videoTrack.getSettings().height,
+    width: videoTrack.getSettings().width,
+  };
+  return cameraInfo;
+}
+
+async function updateConstraintsAndLogInfo(videoDevice, trackEvent) {
+  // See https://developer.mozilla.org/en-US/docs/Web/API/MediaTrackConstraints/facingMode
+  const updatedConstraints = getConstraints(videoDevice.deviceId);
+  try {
+    const stream = await navigator.mediaDevices.getUserMedia(updatedConstraints);
```

Comment:
Are we guaranteed to have permissions on the camera at this point to avoid the user being prompted with a browser permissions prompt?

Reply:
Yes, this is only supposed to run after camera permissions have been granted. We have future work to check camera permission status, but it seemed like an additional layer in an already fairly big PR. [Added a comment ](https://github.com/18F/identity-idp/pull/10227/commits/86fdc971959bfb405c405a2da0bd0a0ebf0b1db6)to explain this to future devs (and future me).